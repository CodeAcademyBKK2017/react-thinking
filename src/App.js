import React, { Component } from 'react';
import Sum from './components/counter/Counter.component'

class App extends Component {
 render() {
   return (
     <Sum/>
   );
 }
}

export default App;
