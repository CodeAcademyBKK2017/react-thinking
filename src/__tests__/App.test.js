import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, {shallow} from 'enzyme';
import App from '../App';
//setup enzyme
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

describe('App', () => {
  it('Overlay: renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('add: should increment state', () => {
    const wrapper = shallow(<App/>);
    const instance = wrapper.instance();
    instance.add();
    expect(instance.state.count).toBe(1);
  });
});
