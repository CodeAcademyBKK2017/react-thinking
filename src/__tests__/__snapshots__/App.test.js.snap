// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Overlay: renders correctly 1`] = `
<div
  className="App"
>
  <input
    type="text"
    value={0}
  />
  <button
    onClick={[Function]}
  >
    ADD
  </button>
  <button
    onClick={[Function]}
  >
    SUBTRACT
  </button>
</div>
`;
